File: rust_bindings/java/src/test/java/app/grafica/twittertext/TestConfiguration.java
package app.grafica.twittertext;

import static org.junit.Assert.*;
import org.junit.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class TestConfiguration {
  static {
    TwitterTextJNILib.load();
  }

  @Test
  public void testConstructor() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertNotNull(ttc);
  }

  @Test
  public void testPath() {
    String path = "rust_bindings/cpp/test_data/test_config.json";
    TwitterTextConfiguration ttc = TwitterTextConfiguration.configurationFromPath(path);
    assertNotNull(ttc);
    assertEquals(ttc.getVersion(), 42);
    assertEquals(ttc.getMaxWeightedTweetLength(), 400);
    assertEquals(ttc.getScale(), 43);
    assertEquals(ttc.getDefaultWeight(), 213);
    assertEquals(ttc.getTransformedUrlLength(), 32);
    WeightedRangeList wrl = ttc.getRanges();
    assertEquals(wrl.size(), 1);
    WeightedRange wr = wrl.get(0);
    assertEquals(wr.getRange().getStart(), 0);
    assertEquals(wr.getRange().getEnd(), 4351);
    assertEquals(wr.getWeight(), 200);
  }

  @Test
  public void testJson() throws IOException {
    String path = "rust_bindings/cpp/test_data/test_config.json";
    byte[] content = Files.readAllBytes(Paths.get(path));
    TwitterTextConfiguration ttc = TwitterTextConfiguration.configurationFromJson(new String(content)); 
    assertNotNull(ttc);
    assertEquals(ttc.getVersion(), 42);
    assertEquals(ttc.getMaxWeightedTweetLength(), 400);
    assertEquals(ttc.getScale(), 43);
    assertEquals(ttc.getDefaultWeight(), 213);
    assertEquals(ttc.getTransformedUrlLength(), 32);
    WeightedRangeList wrl = ttc.getRanges();
    assertEquals(wrl.size(), 1);
    WeightedRange wr = wrl.get(0);
    assertEquals(wr.getRange().getStart(), 0);
    assertEquals(wr.getRange().getEnd(), 4351);
    assertEquals(wr.getWeight(), 200);
  }

  @Test
  public void testVersion() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getVersion(), 3);
    ttc.setVersion(199);
    assertEquals(ttc.getVersion(), 199);
  }

  @Test
  public void testMaxWeightedTweetLength() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getMaxWeightedTweetLength(), 280);
    ttc.setMaxWeightedTweetLength(199);
    assertEquals(ttc.getMaxWeightedTweetLength(), 199);
  }

  @Test
  public void testScale() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getScale(), 100);
    ttc.setScale(199);
    assertEquals(ttc.getScale(), 199);
  }

  @Test
  public void testDefaultWeight() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getDefaultWeight(), 200);
    ttc.setDefaultWeight(199);
    assertEquals(ttc.getDefaultWeight(), 199);
  }

  @Test
  public void testTransformedUrlLength() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getTransformedUrlLength(), 23);
    ttc.setTransformedUrlLength(199);
    assertEquals(ttc.getTransformedUrlLength(), 199);
  }

  @Test
  public void testEmojiParsingEnabled() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    assertEquals(ttc.getEmojiParsingEnabled(), true);
    ttc.setEmojiParsingEnabled(false);
    assertEquals(ttc.getEmojiParsingEnabled(), false);
  }

  @Test
  public void testRanges() {
    TwitterTextConfiguration ttc = new TwitterTextConfiguration();
    WeightedRangeList wrl = ttc.getRanges();
    assertEquals(wrl.size(), 4);
    WeightedRange wr = wrl.get(0);
    assertEquals(wr.getRange().getStart(), 0);
    assertEquals(wr.getRange().getEnd(), 4351);
    assertEquals(wr.getWeight(), 100);
    wr = wrl.get(1);
    assertEquals(wr.getRange().getStart(), 8192);
    assertEquals(wr.getRange().getEnd(), 8205);
    assertEquals(wr.getWeight(), 100);
    wr = wrl.get(2);
    assertEquals(wr.getRange().getStart(), 8208);
    assertEquals(wr.getRange().getEnd(), 8223);
    assertEquals(wr.getWeight(), 100);
    wr = wrl.get(3);
    assertEquals(wr.getRange().getStart(), 8242);
    assertEquals(wr.getRange().getEnd(), 8247);
    assertEquals(wr.getWeight(), 100);
  }

  @Test
  public void testConfigV2() {
    TwitterTextConfiguration ttc = TwitterTextConfiguration.configV2();
    assertEquals(ttc.getVersion(), 2);
    assertEquals(ttc.getEmojiParsingEnabled(), false);
    WeightedRangeList wrl = ttc.getRanges();
    assertEquals(wrl.size(), 4);
  }

  @Test
  public void testConfigV1() {
    TwitterTextConfiguration ttc = TwitterTextConfiguration.configV1();
    assertEquals(ttc.getVersion(), 1);
    assertEquals(ttc.getEmojiParsingEnabled(), false);
    WeightedRangeList wrl = ttc.getRanges();
    assertEquals(wrl.size(), 0);
  }  
}

File: rust/python-bindings/test/test_configuration.py
import os
import pytest
import sys
import twitter_text
import yaml

def test_ctor():
    config = twitter_text.TwitterTextConfiguration()
    assert config is not None
    assert config.get_version() == 3

def test_path():
    path = "rust_bindings/cpp/test_data/test_config.json"
    config = twitter_text.TwitterTextConfiguration.configuration_from_path(path)
    assert config is not None
    assert config.get_version() == 42
    assert config.get_max_weighted_tweet_length() == 400
    assert config.get_scale() == 43
    assert config.get_default_weight() == 213
    assert config.get_transformed_url_length() == 32
    ranges = config.get_ranges()
    assert len(ranges) == 1
    wr = ranges[0]
    assert wr.range.start == 0
    assert wr.range.end == 4351
    assert wr.weight == 200

def test_json():
    with open(r"rust_bindings/cpp/test_data/test_config.json") as file:
        config = twitter_text.TwitterTextConfiguration.configuration_from_json(file.read())
        assert config is not None
        assert config.get_version() == 42
        assert config.get_max_weighted_tweet_length() == 400
        assert config.get_scale() == 43
        assert config.get_default_weight() == 213
        assert config.get_transformed_url_length() == 32
        ranges = config.get_ranges()
        assert len(ranges) == 1
        wr = ranges[0]
        assert wr.range.start == 0
        assert wr.range.end == 4351
        assert wr.weight == 200

def test_accessors():
    config = twitter_text.TwitterTextConfiguration()

    assert config.get_version() == 3
    config.set_version(199)
    assert config.get_version() == 199

    assert config.get_max_weighted_tweet_length() == 280
    config.set_max_weighted_tweet_length(199)
    assert config.get_max_weighted_tweet_length() == 199

    assert config.get_scale() == 100
    config.set_scale(199)
    assert config.get_scale() == 199

    assert config.get_default_weight() == 200
    config.set_default_weight(199)
    assert config.get_default_weight() == 199

    assert config.get_transformed_url_length() == 23
    config.set_transformed_url_length(199)
    assert config.get_transformed_url_length() == 199

    assert config.get_emoji_parsing_enabled() == True
    config.set_emoji_parsing_enabled(False)
    assert config.get_emoji_parsing_enabled() == False

def test_ranges():
    config = twitter_text.TwitterTextConfiguration()
    ranges = config.get_ranges()
    assert ranges[0].range.start == 0
    assert ranges[0].range.end == 4351
    assert ranges[0].weight == 100

    assert ranges[1].range.start == 8192
    assert ranges[1].range.end == 8205
    assert ranges[1].weight == 100

    assert ranges[2].range.start == 8208
    assert ranges[2].range.end == 8223
    assert ranges[2].weight == 100

    assert ranges[3].range.start == 8242
    assert ranges[3].range.end == 8247
    assert ranges[3].weight == 100

def test_v2():
    config = twitter_text.TwitterTextConfiguration.config_v2()
    assert config.get_version() == 2
    assert config.get_emoji_parsing_enabled() == False
    ranges = config.get_ranges()
    assert len(ranges) == 4

def test_v1():
    config = twitter_text.TwitterTextConfiguration.config_v1()
    assert config.get_version() == 1
    assert config.get_emoji_parsing_enabled() == False
    ranges = config.get_ranges()
    assert len(ranges) == 0

if __name__ == "__main__":
    raise SystemExit(pytest.main([__file__]))